### PR TITLE
compiletest: Remove the `-test` suffix from normalize directives

### PR DESCRIPTION
**Pull request by Zalathar**
This suffix was an artifact of using the same condition-checking engine as the `ignore-*` and `only-*` directives, but in practice we have only 2 tests that legitimately use a condition, and both of them only care about 32-bit vs 64-bit.

This PR detaches `normalize-*` directives from the condition checker, and replaces it with a much simpler system of four explicit `NormalizeKind` values. It then takes advantage of that simplicity to get rid of the `-test` suffix.

---

Addresses one of the points of #126372.

The new name-checking code is a bit quaint, but I think it's a definite improvement over the status quo.

---

The corresponding dev-guide update is https://github.com/rust-lang/rustc-dev-guide/pull/2172.

r? jieyouxu

### CODE REVIEW

File: tests/ui/debuginfo/debuginfo-type-name-layout-ice-94961-1.rs [hunk 1]
```diff
@@ -4,8 +4,6 @@
 //@ compile-flags:-C debuginfo=2
 //@ build-fail
 //@ error-pattern: too big for the target architecture
-//@ normalize-stderr-64bit: "18446744073709551615" -> "SIZE"
-//@ normalize-stderr-32bit: "4294967295" -> "SIZE"
```
Comment (Zalathar) [inline tests/ui/debuginfo/debuginfo-type-name-layout-ice-94961-1.rs:8]: Fun fact, this hasn't needed normalization since #106873.

**Comment by Zalathar** [comment]
Archaeological note: The two remaining tests that use conditional normalization are:

- `tests/ui/transmute/main.rs`
- `tests/ui/consts/transmute-size-mismatch-before-typeck.rs`

In both cases, there seems to have been a deliberate attempt to use pointer-size-dependent types, so for now I haven't disturbed them. It's possible that we could just restrict them to `//@ only-64bit` and remove conditional normalizations altogether, but I didn't want to have to justify that extra step in this PR.

**Comment by jieyouxu** [comment]
This is going to take me a bit of time to review (scheduled for tmrw).

**Comment by Zalathar** [comment]
Rebased due to conflicts with #134781 (which added some more normalize directives); no changes.

**Comment by Zalathar** [comment]
Added some FIXMEs for things I noticed but didn't clean up in this PR, since touching hundreds of tests is enough change for one PR [(diff)](https://github.com/rust-lang/rust/compare/9c350ee2bb0dc5995969bf3bcb219d7f96c62181..28788f27dacbcf67225c4683ac6a4b2486cbeeab).

**Review by jieyouxu — APPROVED** [review]
Thanks!

- I reviewed the test diffs manually.
- The normalization rule is a bit wonky still, but that's pre-existing so.

**Comment by jieyouxu** [comment]
This modifies quite a lot of tests, so giving a higher p due to conflict-prone-ness

@bors r+ rollup p=1

**Comment by jieyouxu** [comment]
(It can be rolled up if someone is doing rollups)

**Pull request merged**